src/scalmat.jl

```diff
@@ -187,3 +187,9 @@ function Xt_invA_X(a::ScalMat, x::Matrix{<:Real})
     @check_argdims LinearAlgebra.checksquare(a) == size(x, 1)
     return Symmetric(_rdiv!(transpose(x) * x, a.value))
 end
+
+### dot product
+
+function LinearAlgebra.dot(x::AbstractVector, a::ScalMat, y::AbstractVector)
```

Comment:
We should check that the dimensions match:
```suggestion
function LinearAlgebra.dot(x::AbstractVector, a::ScalMat, y::AbstractVector)
    @check_argdims LinearAlgebra.checksquare(a) == length(x) == length(y)
```